modules/atomic-widgets/props-handler/props-handler.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Elementor\Modules\AtomicWidgets\PropsHandler;
+
+use Elementor\Modules\AtomicWidgets\Schema\Atomic_Prop;
+use Exception;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+class Props_Handler {
+	/**
+	 * Each transformer can return a value that is also a transformable value,
+	 * which means that it can be transformed again by another transformer.
+	 * this constant defines the maximum depth of transformations.
+	 */
+	const TRANSFORM_DEPTH_LIMIT = 3;
+
+	const CONTEXT_SETTINGS = 'settings';
+	const CONTEXT_STYLE = 'style';
+
+	/**
+	 * @var array<Props_Handler>
+	 */
+	private static array $instances = [];
+
+	private Transformers_Registry $transformers;
+
+	public function __construct( Transformers_Registry $transformers ) {
+		$this->transformers = $transformers;
+	}
+
+	public static function instance( string $context, bool $fresh = false ): self {
+		if ( ! isset( self::$instances[ $context ] ) || $fresh ) {
+			$registry = new Transformers_Registry();
+
+			do_action( "elementor/atomic-widgets/{$context}/transformers", $registry );
+
+			self::$instances[ $context ] = new self( $registry );
+		}
+
+		return self::$instances[ $context ];
+	}
```

Comment:
interesting approach

maybe we can add some syntactic sugar:

```PHP
Props_Handler::for_styles()
```

and then the API could be something like:

```PHP
Props_Handler::for_styles()->transform( [
	'props' => $props,
	'schema' => $schema,
] )
```

(not sure about the class name though)

Reply:
Cool Cool Cool, DONE!